### PR TITLE
API Fixed

### PR DESCRIPTION
**Pull request by gonzalogrisafi**
Fixed were made and now code and documentation look a lot better

### CODE REVIEW

**Comment by gonzalogrisafi** [comment]
ps: don't look at app.js

File: grisafi/handler/apphandler.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        }
+    },
+
+    postMember: (req, res) => {
+        console.log("POST /members/")
+        if (req.body.name != null && v.validateId(req.body.id)) {
+            members.push(new Member(req.body.name, req.body.id));
+            res.status(201).json(
+                {
+                    data: {
```
Comment (valdazo) [inline grisafi/handler/apphandler.js:87]: just to keep an uniform result, add the code as part of the returned json.

File: grisafi/handler/apphandler.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                        id: req.body.id,
+                        name: req.body.name
+                    },
+                    message: `Member with id ${req.body.id} created successfully`
+                });
+        }
+        else {
+            res.status(400).json({
+                error: {
+                    code: 404,
```
Comment (valdazo) [inline grisafi/handler/apphandler.js]: change this code to make it the same than the response status.

File: grisafi/handler/apphandler.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+                    message: "Incorrect Parameters"
+                }
+            });
+        }
+    },
+
+    getBooks: (req, res) => {
+        console.log(`GET /books/`);
+        res.status(200).json({ data: books });
```
Comment (valdazo) [inline grisafi/handler/apphandler.js]: decide weather or not you want to put the code as part of the result json. If you want to do it, add it to all the responses.

File: grisafi/handler/apphandler.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
+let f = require('../lib/auxFuncs');
```
Comment (valdazo) [inline grisafi/handler/apphandler.js]: Use a meaningful name instead of f

File: grisafi/handler/apphandler.js [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+let v = require('../lib/validators');
```
Comment (valdazo) [inline grisafi/handler/apphandler.js]: Use a meaningful name instead of v

File: grisafi/app.js [hunk 6]
```diff
@@ -1,10 +1,22 @@
 var express = require('express');
 var app = express();
 var bodyParser = require('body-parser');
+var mysql = require ('mysql');
 app.use(bodyParser.json());
 
 const DIA_EN_MILISEGUNDOS = 1000 * 60 * 60 * 24;
```
Comment (valdazo) [inline grisafi/app.js:7]: This is defined twice, one definition is here, the other definition is in apphandler.js

File: grisafi/lib/auxFuncs.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            return 0;
+        }
+        else {
+            return -1;
+        }
+    },
+
+    //getLoansId returns all the current loans by a member via his id
+    getLoansId: (memberId, loansCollection) => {
+        let prest = new Array();
```
Comment (valdazo) [inline grisafi/lib/auxFuncs.js]: english

File: grisafi/lib/validators.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+    validateId:(id)=>{
+        if(typeof(id)!='number' || id==null){
+            return false;
+        }
+        return true;
+    },
+
+    validateBook: (title, quantity, id)=>{
+        if(/^\d*$/.test(quantity) && /^\d*$/.test(id) && typeof(title)=='string'){
```
Comment (valdazo) [inline grisafi/lib/validators.js]: Shouldn't be /^\d+$/ instead

Comment (valdazo) [inline grisafi/lib/validators.js]: Is an empty string valid aswell?

File: grisafi/handler/apphandler.js [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+                    error: {
+                        code: 404,
+                        message: "book not found"
+                    }
+                })
+            }
+            else if (book.availables() > 0) {
+                loans.push(new Loan(f.generateLoansID(), req.body.memberId, req.body.bookId, req.body.days));
+                res.status(200).json({
+                    success: true,
```
Comment (valdazo) [inline grisafi/handler/apphandler.js]: use one of the many options added.
Either use
code : 200, or success: true, or status:success

File: grisafi/handler/apphandler.js [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+                            message: `there are no available copies of Book ${req.body.bookId} available for loan`
+                        }
+                    })
+            }
+        }
+    },
+
+    deleteLoan: (req, res) => {
+        if (f.returnBook(req.params.id,loans)) {
+            res.status(200).json({
```
Comment (valdazo) [inline grisafi/handler/apphandler.js]: status should be 204

**Review by valdazo — CHANGES_REQUESTED** [review]

File: grisafi/lib/auxFuncs.js (hunk 7 above)
Comment (gonzalogrisafi) [reply grisafi/lib/auxFuncs.js]: my bad

**Review by valdazo — APPROVED** [review]

**Pull request merged**